pkg/patterns/declarative/reconciler.go

```diff
@@ -170,8 +170,8 @@ func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 		}
 
 		if r.options.status != nil {
-			if err = r.options.status.Reconciled(ctx, instance, objects, err); err != nil {
-				log.Error(err, "failed to reconcile status")
+			if statusErr := r.options.status.Reconciled(ctx, instance, objects, err); statusErr != nil {
```

Comment:
_Maybe_ we should return an error, if `err` was nil going in to the defer block, but then we failed to update the status, so we retry again to update the status.  However, I think this fix is likely more important.